go/cmd/dolt/commands/fetch.go

```diff
@@ -77,164 +76,15 @@ func (cmd FetchCmd) Exec(ctx context.Context, commandStr string, args []string,
 	}
 	updateMode := ref.UpdateMode{Force: apr.Contains(cli.ForceFlag)}
 
-	verr := fetchRefSpecs(ctx, updateMode, dEnv, r, refSpecs)
-
-	return HandleVErrAndExitCode(verr, usage)
-}
-
-func getRefSpecs(args []string, dEnv *env.DoltEnv, remotes map[string]env.Remote) (env.Remote, []ref.RemoteRefSpec, errhand.VerboseError) {
-	if len(remotes) == 0 {
-		return env.NoRemote, nil, errhand.BuildDError("error: no remotes set").AddDetails("to add a remote run: dolt remote add <remote> <url>").Build()
-	}
-
-	remName := "origin"
-	remote, remoteOK := remotes[remName]
-
-	if len(args) != 0 {
-		if val, ok := remotes[args[0]]; ok {
-			remName = args[0]
-			remote = val
-			remoteOK = ok
-			args = args[1:]
-		}
-	}
-
-	if !remoteOK {
-		return env.NoRemote, nil, errhand.BuildDError("error: unknown remote").SetPrintUsage().Build()
-	}
-
-	var rs []ref.RemoteRefSpec
 	var verr errhand.VerboseError
```

Comment:
In general, this is an anti-pattern: very easy to just not handle an error when it's declared outside where it's used.

Handle each branch's error separately, returning immediately.